maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/booterclient/JarManifestForkConfiguration.java

```diff
@@ -114,7 +115,7 @@ private File createJar( @Nonnull List<String> classPath, @Nonnull String startCl
             {
                 File file1 = new File( it.next() );
 
-                String uri = parent.relativize( file1.toPath() ).toString();
+                String uri = URI.create( parent.relativize( file1.toPath() ).toString() ).toASCIIString();
```

Comment:
Works for me but the annoying are logs because we log every Jar file.
This can be fixed by using a boolean flag preventing from repeative logs with same message.
Can you segregate the try-catch to a private method?
I have a question regarding the bug. The build plugin crashed with plugin version `2.22.1` on 1.8.0_192? It did not crash on my Windows system. It is different experience on Linux?

Reply:
AFAIK our users, they won't see any warnings. So. We should write the log down to a dump file, see `InPluginProcessDumpSingleton`. You need to specify reports dir and pass it to the class.